tools/engine_tool/lib/src/logger.dart

```diff
@@ -0,0 +1,135 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async' show runZoned;
+import 'dart:io' as io show
+  IOSink,
+  stderr,
+  stdout;
+
+import 'package:logging/logging.dart' as log;
+import 'package:meta/meta.dart';
+
+// This is where a flutter_tool style progress spinner, color output,
+// ascii art, terminal control for clearing lines or the whole screen, etc.
+// can go. We can just add more methods to Logger using the flutter_tool's
+// Logger as a guide:
+//
+// https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/base/logger.dart#L422
```

Comment:
Consider using permalink so that the line number doesn't break:

```suggestion
// https://github.com/flutter/flutter/blob/c530276f7806c77da2541c518a0e103c9bb44f10/packages/flutter_tools/lib/src/base/logger.dart#L422
```